### PR TITLE
add starter plugin version feature (Addresses Issue #166)

### PR DESCRIPTION
**Pull request by skotchpine**
# Plugin Versioning

This isn't a finished product, but a first draft. It adds a versions to `plugin-add` via git branches. With this change, `plguin-add` takes an optional third argument `<git-branch>`. If a branch is given and is not `master`, then the plugin is cloned to `<plugin-name>-<git-branch>` instead of `<plugin-name>`.

### CODE REVIEW

File: lib/commands/plugin-add.sh [hunk 1]
```diff
@@ -28,7 +35,7 @@ plugin_add_command() {
     display_error "Plugin named $plugin_name already added"
     exit 1
   else
-    if ! git clone "$source_url" "$plugin_path"; then
+    if git clone -b "$plugin_branch" "$source_url" "$plugin_path"; then
```
Comment (vic) [inline lib/commands/plugin-add.sh]: I guess you removed the `!` un-intentionally, the script should fail if it's not able to clone the repo.

**Comment by tomdavidson** [comment]
@skotchpine why branches rather than tags?

**Comment by skotchpine** [comment]
@tomdavidson No reason. Why tags rather than branches?

**Comment by tomdavidson** [comment]
Why closed? Vendoring the plugins would be great.

Branches just point to HEAD and are a moving target. In many cases tags vendor strong enough that a separate artifact registry is not needed. SCMs typically have built in features for tags https://github.com/asdf-vm/asdf/releases and there is an whole ecosystems of semver'd git tagged releases.

**Comment by theoretick** [comment]
Any chance we can get this PR re-opened? Or happy to open a fresh one if the original author is unwilling.

Having some basic version locking would be great in order to prevent failures due to plugins being updated. In the meantime we have to use something a bit hackier like `cd`ing to the installation path and checking out a given SHA

**Comment by jthegedus** [comment]
If this were to be reopened, I would like to lookup the default branch of the repo instead of assuming `master` given the recent changes to GitHub et al moving to `main`.

I need to finish #800 before we modify any of this too.

**Comment by jthegedus** [comment]
@theoretick Yes please! Initial install being either remote HEAD, remote <branch> or remote <tag> is (I think) enough dials to satisfy people.

If you want to open a new PR I will happily review, extra points for tests ;)

**Comment by theoretick** [comment]
Sorry for the delay! Opened https://github.com/asdf-vm/asdf/pull/916 as follow-up with a relevant test

**Pull request closed without merging**